### PR TITLE
Carry over query for navigation when opening from SRP

### PR DESCRIPTION
**Pull request by EdwardJTL**
When we open a result from SRP, the new navigation item doesn't get a query value associated in `QueryForNavigation`.
This change carries over the query value, for both opening in place and in a new tab. This way, a page opened from SRP will have the same query value as pages opened from fast tap, which gives better fallback for cheatsheet.

### How was this tested?
- [ ] I tested this manually
- [ ] I added automated tests
- [ ] Existing automated tests are enough to cover my changes

### Manual test cases
Manually checked by opening links from SRP, both in new tab and in place (by changing it in the account settings). Observed that query value is attached and correct.

### CODE REVIEW

**Comment by darinwf** [comment]
Does this interfere with back navigation to the SRP at all?

**Comment by EdwardJTL** [comment]
> Does this interfere with back navigation to the SRP at all?

oh darn, it looks like `Tab.goBack` uses the query to check if it should open the suggestion UI instead... that makes things complicated :/

**Comment by darinwf** [comment]
> > Does this interfere with back navigation to the SRP at all?
> 
> oh darn, it looks like `Tab.goBack` uses the query to check if it should open the suggestion UI instead... that makes things complicated :/

Yeah. that's what i was worried about. It is okay, we just need to add an attribute on the stored query to reflect the mode.

**Comment by EdwardJTL** [comment]
> > > Does this interfere with back navigation to the SRP at all?
> > 
> > 
> > oh darn, it looks like `Tab.goBack` uses the query to check if it should open the suggestion UI instead... that makes things complicated :/
> 
> Yeah. that's what i was worried about. It is okay, we just need to add an attribute on the stored query to reflect the mode.

what should we call this attribute? I believe we need it to indicate whether suggestion UI should be opened for the current navigation item.
or we could have cheatsheet model look for query using the tab's backlist and parent properties, but the parent tab property is not restored on launch, so this won't work sometimes.

**Comment by darinwf** [comment]
> what should we call this attribute? I believe we need it to indicate whether suggestion UI should be opened for the current navigation item. or we could have cheatsheet model look for query using the tab's backlist and parent properties, but the parent tab property is not restored on launch, so this won't work sometimes.

Be creative :) What would you like to call it? I think you have the right idea about what the flag should indicate.

File: Client/Frontend/Browser/Tabs/TabManager/TabManager.swift [hunk 1]
```diff
@@ -349,6 +349,16 @@ class TabManager: NSObject {
             self.selectTab(popup)
         }
 
+        // if we open from SRP, carry over the query
```
Comment (darinwf) [inline Client/Frontend/Browser/Tabs/TabManager/TabManager.swift:352]: So this is for the scenario of a user who has configure neeva to open links in a new tab, right?

**Review by darinwf — APPROVED** [review]

**Review by chung-lai — APPROVED** [review]

File: Client/Frontend/Browser/Tabs/TabManager/TabManager.swift (hunk 1 above)
Comment (EdwardJTL) [reply Client/Frontend/Browser/Tabs/TabManager/TabManager.swift:352]: yes. the other one in `QueryForNavigation` is for opening in place

File: Client/Frontend/Browser/Tabs/QueryForNavigation.swift [hunk 2]
```diff
@@ -3,15 +3,21 @@
 // found in the LICENSE file.
 
 import Foundation
+import Shared
 import WebKit
 
 class QueryForNavigation {
     struct Query {
+        enum Location: Int, Codable {
+            case suggestion
```
[the remark this answers is not in the document]
Comment (EdwardJTL) [reply Client/Frontend/Browser/Tabs/QueryForNavigation.swift:14]: added comments :)

**Pull request merged**